pkg/controllers/provisioning/v1alpha1/reallocator/terminate.go

```diff
@@ -0,0 +1,66 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reallocator
+
+import (
+	"context"
+	"fmt"
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+type Terminator struct {
+	kubeClient client.Client
+}
+
+func (t *Terminator) CordonNodes(ctx context.Context, nodes []*v1.Node) error {
+	for _, node := range nodes {
+		if node.Spec.Unschedulable {
+			continue
+		}
+		persisted := node.DeepCopyObject()
+		node.Spec.Unschedulable = true
+		if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
```

Comment:
Fine for now, but maybe we should use client-go here for such operations.

Reply:
I like the idea of keeping the amount of package dependencies we have on each file to a minimum. Is there a reason why you would like client-go more here?